app/flame_chart/profile_model.ts

```diff
@@ -47,8 +51,16 @@ function eventComparator(a: TraceEvent, b: TraceEvent) {
   return durationDiff;
 }
 
-function threadNameComparator(a: ThreadTimeline, b: ThreadTimeline) {
-  // Sort by thread name.
+function timelineComparator(a: ThreadTimeline, b: ThreadTimeline) {
+  // Within numbered thread names (e.g. "skyframe evaluator 0", "grpc-command-0"), sort
+  // numerically.
+  const matchA = a.threadName.match(NUMBERED_THREAD_NAME_PATTERN)?.groups;
+  const matchB = b.threadName.match(NUMBERED_THREAD_NAME_PATTERN)?.groups;
+  if (matchA && matchB && matchA["prefix"] === matchB["prefix"]) {
```

Comment:
think it's a little more readable as 
```
if (matchA && matchB && matchA.prefix === matchB.prefix) {
  return Number(matchA.number) - Number(matchB.number);
}
```

Reply:
woops, missed this. Will clean up the next time I pass through here